pkg/lakom/resolvetag/resolvetag_suite_test.go

```diff
@@ -5,13 +5,207 @@
 package resolvetag_test
 
 import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"log"
+	"net/http/httptest"
+	"net/url"
+	"os"
 	"testing"
 
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/registry"
+	registryv1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/random"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+	"github.com/sigstore/cosign/v2/pkg/cosign"
+	"github.com/sigstore/cosign/v2/pkg/oci/mutate"
+	ociRemote "github.com/sigstore/cosign/v2/pkg/oci/remote"
+	"github.com/sigstore/cosign/v2/pkg/oci/signed"
+	"github.com/sigstore/cosign/v2/pkg/oci/static"
+	"github.com/sigstore/sigstore/pkg/signature"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+var (
+	// Fake registry properties
+	registryURL *url.URL
+	server      *httptest.Server
+
+	signedImageFullRef   string
+	unsignedImageFullRef string
+
+	signedImageTagRef      string
+	unsignedImageTagRef    string
+	nonExistantImageTagRef string
+
+	// Key for signing images in fake registry
+	privateKey *ecdsa.PrivateKey
+
+	// Public key in PEM format for verifying signatures in the fake registry
+	publicKey string
+)
+
+const (
+	signedImageTag   = "signed"
+	unsignedImageTag = "unsigned"
 )
 
 func TestCMD(t *testing.T) {
 	RegisterFailHandler(Fail)
 	RunSpecs(t, "ResolveTag Suite")
 }
+
+var _ = BeforeSuite(func() {
+	scheme = runtime.NewScheme()
+	err := corev1.AddToScheme(scheme)
+	Expect(err).ToNot(HaveOccurred())
+
+	dirPath, err := os.MkdirTemp("", "verifysignature_test")
+	Expect(err).ToNot(HaveOccurred())
+	DeferCleanup(func() {
+		err := os.RemoveAll(dirPath)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	// Tests rely on a fake registry
+	registryURL, server = startFakeRegistry()
+	DeferCleanup(func() {
+		server.Close()
+	})
+
+	signedImage, signedImageRef, err := writeRandomImage(signedImageTag)
+	Expect(err).ToNot(HaveOccurred())
+	_, unsignedImageRef, err := writeRandomImage(unsignedImageTag)
+	Expect(err).ToNot(HaveOccurred())
+
+	signedImageFullRef = signedImageRef.Name()
+	unsignedImageFullRef = unsignedImageRef.Name()
+
+	signedImageTagRef = signedImageRef.Context().Tag(signedImageTag).String()
+	unsignedImageTagRef = unsignedImageRef.Context().Tag(unsignedImageTag).String()
+
+	nonExistantImageTagRef = fmt.Sprintf("%s:nonexistant", signedImageRef.Context().Name())
+
+	privateKey, err = cosign.GeneratePrivateKey()
+	Expect(err).ToNot(HaveOccurred())
+	publicKey, err = publicKeyToPEM(privateKey.Public())
+	Expect(err).ToNot(HaveOccurred())
+
+	err = signImage(signedImage, signedImageFullRef)
+	Expect(err).ToNot(HaveOccurred())
+})
+
+var _ = AfterSuite(func() {
+	server.Close()
```

Comment:
From the `DeferCleanup` docs

> When DeferCleanup is called in BeforeSuite, SynchronizedBeforeSuite, AfterSuite, or SynchronizedAfterSuite the registered callback will be invoked when the suite completes (i.e. it will behave like an AfterSuite node)

I think you can skip the `AfterSuite` node. Then you do not need the `server` variable to be in the global scope, it can be a local to the `BeforeSuite` node scope.